BEIMA.Backend/DeviceFunctions/DeleteDevice.cs

```diff
@@ -35,6 +49,25 @@ public static class DeleteDevice
 
             var mongo = MongoDefinition.MongoInstance;
             var deviceId = new ObjectId(id);
+            var deviceDocument = mongo.GetDevice(deviceId);
+            if(deviceDocument == null)
+            {
+                return new NotFoundObjectResult(Resources.DeviceNotFoundMessage);
+            }
+
+            var device = BsonSerializer.Deserialize<Device>(deviceDocument);
+
+            // Remove Files
+            foreach (var file in device.Files)
+            {
+                await _storage.DeleteFile(file.FileUid);
+                // Log to file if delete failed
```

Comment:
```suggestion
                log.LogInformation($"Failed to delete a file: {file.FileUid.ToString()}.");
```

Reply:
Wasn't sure if we wanted to use the logger or if we wanted to write all the failures to a file that someone could then open and read through.

Reply:
a file sounds good so they can be reviewed, a "view logs" page can be added to the suggested improvements too
alternatively we could log to the DB